src/profile/public/main.js

```diff
@@ -454,46 +381,25 @@ Promise.all([
 		event.preventDefault();
 		const button = $(this).find('button[type="submit"]');
 		disableSubmitButton(button);
-		const encoder = new TextEncoder();
-		const nonce = new Uint8Array(8);
-		const messageType = new Uint8Array([1]);
-		const firstName = encoder.encode($(this).find('#first-name').val());
-		const firstNameLen = firstName.length;
-		const lastName = encoder.encode($(this).find('#last-name').val());
-		const lastNameLen = lastName.length;
-		const title = encoder.encode($(this).find('#title').val());
-		const titleLen = title.length;
-		const company = encoder.encode($(this).find('#company').val());
-		const companyLen = company.length;
-		const experience = encoder.encode($(this).find('#experience').val());
-		const experienceLen = experience.length;
-		var message = new Uint8Array(15 + firstNameLen + lastNameLen + titleLen + companyLen + experienceLen);
-		message.set(nonce);
-		message.set(messageType, 8);
-		message.set(new Uint8Array([firstNameLen]), 9);
-		message.set(firstName, 10);
-		message.set(new Uint8Array([lastNameLen]), 10 + firstNameLen);
-		message.set(lastName, 11 + firstNameLen);
-		message.set(new Uint8Array([titleLen]), 11 + firstNameLen + lastNameLen);
-		message.set(title, 12 + firstNameLen + lastNameLen);
-		message.set(new Uint8Array([companyLen]), 12 + firstNameLen + lastNameLen + titleLen);
-		message.set(company, 13 + firstNameLen + lastNameLen + titleLen);
-		console.log(experienceLen);
-		message.set(new Uint8Array([experienceLen / 256]), 13 + firstNameLen + lastNameLen + titleLen + companyLen);
-		message.set(new Uint8Array([experienceLen % 256]), 14 + firstNameLen + lastNameLen + titleLen + companyLen);
-		message.set(experience, 15 + firstNameLen + lastNameLen + titleLen + companyLen);
-		const signature = nacl.sign(message, keyPair.secretKey);
-		const signer = keyPair.publicKey;
```

Comment:
Since we can rely on the UserLib to construct the signed message, a lot of this becomes unnecessary.